### ISSUE TITLE
python3-sys and PYTHON_SYS_EXECUTABLE shadows system python  

### ISSUE THREAD

**Issue by svevang** [dgrunwald/rust-cpython#135]
When determining which python to use, you can force rust-cpython to use a particular python by setting `PYTHON_SYS_EXECUTABLE` and the appropriate feature. For instance, if I want to use python 3.5 with a custom interpreter location. I can:

```
PYTHON_SYS_EXECUTABLE="/some/path/bin/python3" cargo build --features "python-3-5" -vv
``` 

Setting the cargo feature will branch compilation of the rust-cpython code. The build.rs script verifies that the particular interpreter we've specified in `PYTHON_SYS_EXECUTABLE` is compatible with the feature we've picked.

However, when we're determining how to link to python, we're not using `PYTHON_SYS_EXECUTABLE`:
https://github.com/dgrunwald/rust-cpython/blob/master/python3-sys/build.rs#L200
This means a system python in our PATH will shadow the python we want to link.

Steps to reproduce:

First, ensure that `python` in your path is statically linked.

Next build python as a framework, and attempt to link it to rust-cpython

```
cd /tmp
git clone https://github.com/dgrunwald/rust-cpython.git

mkdir -p /tmp/rust-cpython/python-dist/framework
mkdir -p /tmp/rust-cpython/python-dist/src

cd /tmp/rust-cpython/python-dist/src
curl https://www.python.org/ftp/python/3.5.3/Python-3.5.3.tgz | tar -xvf -
cd Python-3.5.3
./configure --prefix=/tmp/rust-cpython/python-dist/framework --enable-framework --libdir=/tmp/rust-cpython/python-dist/framework/lib --enable-framework=/tmp/rust-cpython/python-dist/framework
make -j8 
make install

cd /tmp/rust-cpython/
# cargo build --features "python-3" -vv
PYTHON_SYS_EXECUTABLE="/tmp/rust-cpython/python-dist/framework/Python.framework/Versions/Current/bin/python3" cargo build --features "python-3-5" -vv 

```
Expected results: rust-cpython's build artifacts should link to our specially compiled python framework.

Actual results: The following error indicates that we're trying to statically link the framework (rather than dynamically linking it.)
```
cargo:python_interpreter=/private/tmp/rust-cpython/python-dist/framework/Python.framework/Versions/3.5/bin/python3
     Running `rustc --crate-name python3_sys python3-sys/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 --cfg 'feature="default"' --cfg 'feature="python-3"' --cfg 'feature="python-3-5"' -C metadata=c18834b5baee6b7d -C extra-filename=-c18834b5baee6b7d --out-dir /private/tmp/rust-cpython/target/debug/deps -C incremental=/private/tmp/rust-cpython/target/debug/incremental -L dependency=/private/tmp/rust-cpython/target/debug/deps --extern libc=/private/tmp/rust-cpython/target/debug/deps/liblibc-6abd90f68c9e113d.rlib -L native=/tmp/rust-cpython/python-dist/framework/lib --cfg Py_3_4 --cfg Py_3_5 --cfg 'py_sys_config="WITH_THREAD"' -l static=python3.5m`
     Running `/private/tmp/rust-cpython/target/debug/build/cpython-9c8c9bd9028640ed/build-script-build`
cargo:rustc-cfg=py_sys_config="WITH_THREAD"
error: could not find native static library `python3.5m`, perhaps an -L flag is missing?

error: Could not compile `python3-sys`.

Caused by:
  process didn't exit successfully: `rustc --crate-name python3_sys python3-sys/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 --cfg feature="default" --cfg feature="python-3" --cfg feature="python-3-5" -C metadata=c18834b5baee6b7d -C extra-filename=-c18834b5baee6b7d --out-dir /private/tmp/rust-cpython/target/debug/deps -C incremental=/private/tmp/rust-cpython/target/debug/incremental -L dependency=/private/tmp/rust-cpython/target/debug/deps --extern libc=/private/tmp/rust-cpython/target/debug/deps/liblibc-6abd90f68c9e113d.rlib -L native=/tmp/rust-cpython/python-dist/framework/lib --cfg Py_3_4 --cfg Py_3_5 --cfg py_sys_config="WITH_THREAD" -l static=python3.5m` (exit code: 101)
```

**Comment by fanzeyi** [comment]
I run into the same problem today when compiling python27-sys on my laptop. I have a Python 2.7 compiled statically with `pyenv` so the `build.rs` was tricked to try to link statically with another copy of Python 2.7 that is dynamically linked. 

cc @markbt

**Comment by markbt** [comment]
I think I fixed this in #202. Let me know if not.
